reactor-core/src/main/java/reactor/core/scheduler/BoundedElasticScheduler.java

```diff
@@ -204,6 +210,25 @@ public void dispose() {
 		}
 	}
 
+	@Override
+	public Mono<Void> disposeGracefully(Duration gracePeriod) {
+		BoundedServices services = BOUNDED_SERVICES.get(this);
+		if (services != SHUTDOWN && BOUNDED_SERVICES.compareAndSet(this, services, SHUTDOWN)) {
+			ScheduledExecutorService e = EVICTOR.getAndSet(this, null);
+			Mono<Void> evictorDone;
+			if (e != null) {
+				Sinks.Empty<Void> done = Sinks.empty();
+				Schedulers.shutdownAndAwait(e, gracePeriod, done);
```

Comment:
at one point I was thinking that these `EVICTOR` threads in the various pooling schedulers would be a good candidate to run the `awaitTermination` on. that said, it doesn't scale well since `awaitTermination` is blocking: a pool of 3 workers each with their own pending task of 2s would take 6s to terminate gracefully this way, but 2s only if each await is run on a dedicated thread.